vllm/v1/worker/tpu_model_runner.py

```diff
@@ -653,10 +659,12 @@ def _execute_mm_encoder(self, scheduler_output: "SchedulerOutput"):
             # 2. A list or tuple (length: num_items) of tensors, each of shape
             # (feature_size, hidden_size) in case the feature size is dynamic
             # depending on the input multimodal items.
-            xm.mark_step()
+            if os.environ.get("VLLM_TORCHAX_ENABLED", "0") == "0":
```

Comment:
I think to avoid the clutter we could just define some mark_step() util that does this check internally so we don't add N lines.